cli/engine/providers/azure/APIProxy.py

```diff
@@ -59,7 +59,10 @@ def get_ips_for_feature(self, component_key):
         for instance in running_instances:
             if isinstance(instance, list):
                 instance = instance[0]
-            name = instance['virtualMachine']['name']
+            if self.cluster_model.specification.cloud.hostname_domain_extension != '':
+                name = instance['virtualMachine']['name'] + f'.{self.cluster_model.specification.cloud.hostname_domain_extension}'
+            else:
+                name = instance['virtualMachine']['name']
```

Comment:
Could be something like this, but up to you
```
name = instance['virtualMachine']['name']
if self.cluster_model.specification.cloud.hostname_domain_extension != '':
  name += f'.{self.cluster_model.specification.cloud.hostname_domain_extension}'
```

or

```
name = (instance['virtualMachine']['name'] + f'.{self.cluster_model.specification.cloud.hostname_domain_extension}').rstrip('.')
```

Reply:
Actually your first suggestion was my initial version, but as this for is a weird edge case in behavior I went with the normal if-statement to make it more readable and easier to understand. 

As for you second suggestion, yes I like it, but again not in the context;) 

IMO leave as is.